ctypesgen/includes.py

```diff
@@ -0,0 +1,44 @@
+import re
+import mmap
+
+
+INCLUDE_RE = re.compile(rb'^\s*\#include\s+(["<][^">]+[">])', re.MULTILINE)
```

Comment:
This RE was based on considerations from https://stackoverflow.com/questions/1420017/regular-expression-to-match-c-include-file/
But revisiting, I think it would be better to use to separate capture strategies because we need matching `""` or `<>`, but not `">` or `<"`. Also it is theoretically possible to do `"_<_>_"` or `<_"_>`

Reply:
Improved regex:
`re.compile(rb'^\s*\#include\s+((<[^>]+>)|("[^"]+"))', re.MULTILINE)`

Reply:
Then we get a layout like `(b'"pymacconfig.h"', b'', b'"pymacconfig.h"'), (b'<assert.h>', b'<assert.h>', b'')`, where we can easily determine whether it's a system or relative include.

Reply:
Even better: use a non-capturing wrapper group to save space:
`rb'^\s*\#include\s+(?:<([^>]+)>|"([^"]+)")'`
which results in `(b'', b'pymacconfig.h'), (b'assert.h', b'')`

Reply:
This also matches newlines, which is questionable:
```pycon
>>> data = b"""
... #include <ab
... cd>
... """
>>> import re
>>> INCLUDE_RE = re.compile(rb'^\s*\#include\s+(?:<([^>]+)>|"([^"]+)")', re.MULTILINE)
>>> INCLUDE_RE.findall(data)
[(b'ab\ncd', b'')]
```
On the other hand, it doesn't really matter how we behave here because it's invalid anyway.

Reply:
Maybe we should change the `[^...]` to just `.+`, like so:
`re.compile(rb'^\s*\#include\s+(?:<(.+)>|"(.+)")', re.MULTILINE)`

Reply:
Interestingly, `[m.groups() for m in INCLUDE_RE.finditer(data)]` behaves differently than `INCLUDE_RE.findall(data)`. The former returns `None` for unmatched groups, whereas the latter returns an empty byte string.
`None` is more precise because formally there's a difference between an unmatched group and an empty match (consider `#include ""` vs `#include <>`). But as this is invalid, it's not covered by the RE anyway (but it would if we had `.*` instead of `.+`).